django_admin_fast_search/admin.py

```diff
@@ -41,6 +45,40 @@ def get_list_filter(self, request):
             new_filters += (filter_class,)
         return new_filters + initial_list_filters
 
+    def get_paginator(self, request, *args, **kwargs):
+        """Return an approximate count on pagination only if no filters are specified"""
+        if request.GET:
+            return super().get_paginator(request, *args, **kwargs)
+        return ApproximatePaginator(*args, **kwargs)
+
+
+class ApproximatePaginator(Paginator):
+
+    @cached_property
+    def count(self):
+        """Return a faster approximate row count for MySQL, otherwise fallback to default behavior"""
+        connection = connections[self.object_list.db]
+        is_mysql = getattr(connection, 'vendor', None) == 'mysql'
+
+        if is_mysql and isinstance(self.object_list, query.QuerySet):
+            return self._mysql_approximate_count(connection)
+
+        return super().count
+
+    def _mysql_approximate_count(self, connection):
+        """Use MySQL information_schema to get an approximate row count on the table"""
+        with connection.cursor() as cursor:
+            table_name = self.object_list.model._meta.db_table
+            cursor.execute(
```

Comment:
This seems like it wouldn't work if there's a filter active? Maybe there should be a check of whether it can be estimated like in `django-mysql` [here](https://github.com/adamchainz/django-mysql/blob/f534fddefb184e7ae1ec3f1fc6c4139c0677562b/src/django_mysql/models/query.py#L705)?

Reply:
Yes that is correct though that case is handled by `get_paginator()` and `ApproximatePaginator` will never be instantiated if there are active filters. 

https://github.com/Instawork/django-admin-fast-search/pull/13/files#diff-be14cbf81a7f63f1f3bfbdd22b90d8a2d7fa87fe18dbbf6474f57b54b5815e94R50

Considering this is for `admin.py` functionality I wasn't imaging the approximator paginator to be used anywhere else. Perhaps a few options:

1) Leave as-is.
2) Rename `class ApproximatePaginator` to `class _ApproximatePaginator` to hint that it shouldn't be used elsewhere.
3) Introduce the logic you proposed.  

Option 3) felt a little duplicative given we're checking `request.GET` already but I think your callout is totally reasonable. Are you leaning strongly in one particular direction?

Reply:
I may be misunderstanding something, wouldn't this method also be called if you're searching for something in Django admin? And then it would just return the approximate total rows from the table?

I think it's unlikely for this to be used anywhere else - option 3 is definitely the preference.

Reply:
Thanks @dnbedwards. I may be misinterpreting some of the internals on the django admin side but I liked your suggestion and it felt easy enough to incorporate the check on estimate (similar to how `django-mysql` performs it)

ff358b78b0545ff72b8f496e39d19c823b6aa22b